### DISCUSSION TITLE
用类似插件的方式实现支持Hysteria2

### DISCUSSION THREAD

**Discussion by 2dust** [2dust/v2rayNG#3605] [Polls]
本App一直没有支持Hysteria2 协议，是因为使用的Core没有支持
最近从从别的App抄了作业，用类似插件的方式简单支持了

**Comment by 2dust** [comment]
内置插件的测试版本  

https://github.com/2dust/v2rayNG/releases/download/1.9.3/zz-v2rayNG_1.9.3-test3_arm64-v8a.apk

**Reply by 2dust** [reply]
没有看到启动hy2的日志，你在关于里面看下版本号

**Reply by 2dust** [reply]
这个日志看都已经正常运行了
请问"tls":{"sni":" MyIP"}}，这里你是IP还是域名？证书是自签还是？

**Reply by YUEWANGXIAOBAI** [reply]
这里显示的是我的hysteria2服务器IP，证书是在hysteria2配置acme自签
我的hysteria2服务器配置是这样的：
listen: :2087

acme:
  domains: Myhost
  email: Myemail

auth:
  type: password
  password: Mypassword

masquerade:
  type: proxy
  proxy:
    url: https://www.bilibili.com/
    rewriteHost: true

**Reply by 2dust** [reply]
https://github.com/2dust/v2rayNG/releases/download/1.9.3/zz-v2rayNG_1.9.3-test3_arm64-v8a.apk

在配置界面，把跳过证书设置true，试一试

**Reply by 2dust** [reply]
你可把v2rayN那边的配置截图下，脱敏

**Reply by YUEWANGXIAOBAI** [reply]
好了，没有把跳过证书设置成true，也可以正常使用了，谢谢，现在还有个小问题，就是PC和手机地址栏都是用的IP而不是域名，VLESS配置的时候用域名都可以，hysteria2用域名就无法连接
![Screenshot_2024-09-27-15-46-14-926_com v2ray ang-](https://github.com/user-attachments/assets/fe725aee-5c19-4e2a-bf26-2b73508eaf5f)

**Reply by 2dust** [reply]
理论上用域名也是一样的，因为会被解析。不过用ip其实更好些，少了解析。先用着吧
多谢反馈

**Reply by YUEWANGXIAOBAI** [reply]
哦哦，好的，客气了

**Comment by panluoyant** [comment]
现在最新的v2fly core支持hy2了

**Reply by 2dust** [reply]
是的，xray还不支持
